VideoAPI/Video.API/Controllers/AudioRecordingController.cs

```diff
@@ -214,19 +222,30 @@ public async Task<IActionResult> DeleteAudioStreamAsync(Guid hearingId)
         public async Task<IActionResult> GetAudioRecordingLinkAsync(Guid hearingId)
         {
             _logger.LogInformation($"Getting audio recording link for hearing: {hearingId}");
-
             var filePath = $"{hearingId}.mp4";
-            
-            if (!await _storageService.FileExistsAsync(filePath))
-            {
-                _logger.LogWarning($"Audio recording file not found for hearing: {hearingId}");
 
+            var result = await CheckAudioRecordingFile(hearingId);
+            if (!result)
+            {
                 return NotFound();
             }
-
+            
             var audioFileLink = await _storageService.CreateSharedAccessSignature(filePath, TimeSpan.FromDays(14));
-
             return Ok(new AudioRecordingResponse {AudioFileLink = audioFileLink});
         }
+
+         private async Task<bool> CheckAudioRecordingFile(Guid hearingId)
```

Comment:
If you change this to EnsureAudioFileExists and that THROWS an audioplatformexception, that would be better.

Reply:
renamed method